apis/v1alpha1/gateway_types.go

```diff
@@ -34,18 +34,25 @@ import (
 // This ensures that a GatewayClass associated with a Gateway(s) is not
 // deleted while in use.
 type Gateway struct {
-	metav1.TypeMeta   `json:",inline"`
+	metav1.TypeMeta `json:",inline"`
+	// +optional
```

Comment:
is object meta really optional?

Reply:
Yes, according to k8s api conventions, any field with `omitempty` should be marked as `+optional`.

xref: https://github.com/kubernetes/api/blob/master/apps/v1/types.go

Reply:
Are there any object that do this? I don't recall seeing it before.

Reply:
I have not seen any object omit this field.